metadetect/lsst_measure.py

```diff
@@ -76,50 +129,74 @@ def measure_weighted_moments(
 
             ormask = ormasks[i]
 
-            # This will insert a single source into the image
-            replacer.insertSource(source.getId())
+            if use_deblended_stamps:
+                # This will insert a single source into the image
+                replacer.insertSource(source.getId())
 
-            meas_task.callMeasure(source, exposure)
+            if meas_task is not None:
+                meas_task.callMeasure(source, exposure)
 
-            # TODO variable box size
+            # TODO variable box size?
             stamp_bbox = _get_bbox_fixed(
                 exposure=exposure,
                 source=source,
-                stamp_size=48,
+                stamp_size=stamp_size,
             )
             subim = _get_padded_sub_image(exposure=exposure, bbox=stamp_bbox)
 
-            obs = _extract_obs(
-                subim=subim,
-                source=source,
-            )
+            obs = _extract_obs(subim=subim, source=source)
 
-            pres = _measure_moments(obs=obs.psf, weight=weight)
-            ores = _measure_moments(obs=obs, weight=weight)
+            pres = _measure_one(obs=obs.psf, fitter=fitter)
+            ores = _measure_one(obs=obs, fitter=fitter)
 
             res = _get_output(
-                source=source, res=ores, pres=pres, ormask=ormask,
-                box_size=obs.image.shape[0],
-                exp_bbox=exp_bbox,
+                fitter=fitter, source=source, res=ores, pres=pres, ormask=ormask,
+                box_size=obs.image.shape[0], exp_bbox=exp_bbox,
             )
 
-            # Remove object
-            replacer.removeSource(source.getId())
+            if use_deblended_stamps:
+                # Remove object
+                replacer.removeSource(source.getId())
 
             results.append(res)
 
-        # Insert all objects back into image
-        replacer.end()
+        if use_deblended_stamps:
+            # Insert all objects back into image
+            replacer.end()
```

Comment:
Do we need some sort of `try...finally` block to ensure `replacer.end` is always called?

Reply:
good point; it looks like the replacer does not run end() when it is cleaned up